worker/request_handlers.go

```diff
@@ -70,24 +97,48 @@ func HandleSelfStreamRecordEnd(ctx *StreamContext) {
 	notifySilenceResults(sd.Silences, ctx.streamId)
 }
 
-//HandleSelfStreamEnd stops the ffmpeg instance by sending a SIGINT to it and prevents the loop to restart it by marking the stream context as stopped.
-func HandleSelfStreamEnd(ctx *StreamContext) {
+func HandleStreamEndRequest(request *pb.EndStreamRequest) {
+	log.Info("Attempting to end stream: ", request.StreamID)
+	mutex.Lock()
```

Comment:
When working with mutexes it's often best to call `defer mutex.Unlock()` right after locking it to make sure resources are available even if the method aborts unexpectedly.
This is a great resource: https://github.com/uber-go/guide/blob/master/style.md#defer-to-clean-up

Reply:
Yeah, `defer` seems pretty cool for locks! I think I got an idea how I can make the Locks even nicer now :+1:

Reply:
Yup, that looks great now :+1: